ddtrace/internal/datadog/profiling/dd_wrapper/test/forking.cpp

```diff
@@ -0,0 +1,150 @@
+#include "interface.hpp"
+#include "test_utils.hpp"
+#include <gtest/gtest.h>
+
+#include <chrono>
+
+// Initiate an upload in a separate thread, otherwise we won't be mid-upload during fork
+void
+upload_in_thread()
+{
+    std::thread([&]() { ddup_upload(); }).detach();
+}
+
+[[noreturn]] void
+profile_in_child(unsigned int num_threads, unsigned int run_time_ns, std::atomic<bool>& done)
+{
+    // Assumes setup has been called. Launch some samplers, wait, upload, return.
+    // Exit because we don't want this process to exit
```

Comment:
Do you mean "to fork"?

Reply:
Uh, actually I meant "exit, because we want execution within this process to remain confined to this function."  Will clarify